java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,656 @@
+        build(e.leftOperand());
+        break;
+      }
+      case CONDITIONAL_OR: {
+        BinaryExpressionTree e = (BinaryExpressionTree) tree;
+        // process RHS
+        Block trueBlock = currentBlock;
+        currentBlock = createBlock(trueBlock);
+        build(e.rightOperand());
+        Block falseBlock = currentBlock;
+        // process LHS
+        currentBlock = createBranch(e, trueBlock, falseBlock);
+        build(e.leftOperand());
+        break;
+      }
+      case LABELED_STATEMENT: {
+        LabeledStatementTree s = (LabeledStatementTree) tree;
+        build(s.statement());
+        currentBlock = createBlock(currentBlock);
+        labels.put(s.label().name(), currentBlock);
+        break;
+      }
+      case SWITCH_STATEMENT: {
+        // FIXME useless node created for default cases.
+        SwitchStatementTree switchStatementTree = (SwitchStatementTree) tree;
+        Block switchSuccessor = currentBlock;
+        // process condition
+        currentBlock = createBlock();
+        currentBlock.terminator = switchStatementTree;
+        switches.addLast(currentBlock);
+        build(switchStatementTree.expression());
+        // process body
+        currentBlock = createBlock(switchSuccessor);
+        breakTargets.addLast(switchSuccessor);
+        if (!switchStatementTree.cases().isEmpty()) {
+          CaseGroupTree firstCase = switchStatementTree.cases().get(0);
+          for (CaseGroupTree caseGroupTree : switchStatementTree.cases()) {
+            build(caseGroupTree.body());
+            switches.getLast().successors.add(currentBlock);
+            if (caseGroupTree != firstCase) {
+              // No block predecessing the first case group.
+              currentBlock = createBlock(currentBlock);
+            }
+          }
+        }
+        breakTargets.removeLast();
+        // process condition
+        currentBlock = switches.removeLast();
+        break;
+      }
+      case BREAK_STATEMENT: {
+        BreakStatementTree breakStatementTree = (BreakStatementTree) tree;
+        if(breakStatementTree.label() == null) {
+          if (breakTargets.isEmpty()) {
+            throw new IllegalStateException("'break' statement not in loop or switch statement");
+          }
+          currentBlock = createUnconditionalJump(tree, breakTargets.getLast());
+        } else {
+          currentBlock = createUnconditionalJump(tree, null);
+          gotos.add(currentBlock);
+        }
+        break;
+      }
+      case CONTINUE_STATEMENT: {
+        ContinueStatementTree continueStatementTree = (ContinueStatementTree) tree;
+        if(continueStatementTree.label() == null) {
+          if (continueTargets.isEmpty()) {
+            throw new IllegalStateException("'break' statement not in loop or switch statement");
+          }
+          currentBlock = createUnconditionalJump(tree, continueTargets.getLast());
+        } else {
+          currentBlock = createUnconditionalJump(tree, null);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 13. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)